### PR TITLE
[Sliding Expiration] Remove custom model to store data and use Hashes

### PR DESCRIPTION
**Pull request by timaxapa**
Revert back storing data in the Hash
image

### CODE REVIEW

File: src/Cache/RedisDistributedCache.cs [hunk 1]
```diff
@@ -153,22 +172,21 @@ public void Set<T>(string key, T value, DistributedCacheEntryOptions options)
...
                 var expiresIn = this.GetExpirationInSeconds(creationTime, absoluteExpiration, options);
 
                 if (expiresIn.HasValue)
                 {
-                    client.Set(key, entry, TimeSpan.FromSeconds(expiresIn.Value));
-                }
-                else
-                {
-                    client.Set(key, entry);
+                    typedClient.ExpireEntryIn(key, TimeSpan.FromSeconds(expiresIn.Value));
```
Comment (vasont) [inline src/Cache/RedisDistributedCache.cs:189]: is expiresIn.Value correct? if (expiresIn.HasValue) == false else ...

File: src/Cache/RedisDistributedCache.cs [hunk 2]
```diff
@@ -71,15 +93,12 @@ public T Get<T>(string key)
...
-                if (client.ContainsKey(key))
-                {
-                    var entry = client.Get<RedisDistributedCacheEntry<T>>(key);
-
-                    if (entry != null)
-                    {
-                        result = entry.Value;
-                    }
+                var typedClient = client.As<T>();
+                var redisHash = typedClient.GetHash<string>(key);
```
Comment (vasont) [inline src/Cache/RedisDistributedCache.cs:97]: should call this after the if (typedClient.ContainsKey(key)) ?

File: src/Cache/RedisDistributedCache.cs [hunk 3]
```diff
@@ -272,7 +294,7 @@ public void Remove(string key)
         /// <returns>Returns the task to execute.</returns>
         public Task RemoveAsync(string key, CancellationToken token = default)
         {
-            return Task.Run(() => this.Remove(key), token);
+            return Task.Run(() => Remove(key), token);
```
Comment (vasont) [inline src/Cache/RedisDistributedCache.cs:297]: Unless the method is static, StyleCop rule is to include this.

**Review by RobK410 — COMMENTED** [review]
Added some comments for code position and style.

File: src/Cache/RedisDistributedCache.cs (hunk 1 above)
Comment (timaxapa) [reply src/Cache/RedisDistributedCache.cs:189]: Yes. `expiresIn` is a nullable number of seconds for initial expiration

File: src/Cache/RedisDistributedCache.cs (hunk 2 above)
Comment (timaxapa) [reply src/Cache/RedisDistributedCache.cs:97]: Yes. Will move it to deeper scope

File: src/Cache/RedisDistributedCache.cs (hunk 3 above)
Comment (timaxapa) [reply src/Cache/RedisDistributedCache.cs:297]: FIxed all StyleCop rules, except of _type in word Redis_

**Review by RobK410 — APPROVED** [review]

**Pull request merged**